src/halmos/__main__.py

```diff
@@ -393,38 +403,50 @@ def run(
 
     # check assertion violations
     normal = 0
-    models = []
+    execs_to_model = []
+    models: List[ModelWithContext] = []
     stuck = []
+
     for idx, ex in enumerate(exs):
         if args.debug: print(f'Checking output: {idx+1} / {len(exs)}')
 
         opcode = ex.current_opcode()
         if opcode in [EVM.STOP, EVM.RETURN]:
             normal += 1
         elif opcode in [EVM.REVERT, EVM.INVALID]:
-            # Panic(1) # bytes4(keccak256("Panic(uint256)")) + bytes32(1)
-            if ex.output == int('4e487b71' + '0000000000000000000000000000000000000000000000000000000000000001', 16): # 152078208365357342262005707660225848957176981554335715805457651098985835139029979365377
-                gen_model(args, models, idx, ex)
+            # Panic(1)
+            # bytes4(keccak256("Panic(uint256)")) + bytes32(1)
+            if args.debug: print(f'  Will generate model for Panic(1)')
+            if ex.output == 0x4e487b710000000000000000000000000000000000000000000000000000000000000001:
+                execs_to_model.append((idx, ex))
         elif ex.failed:
-            gen_model(args, models, idx, ex)
+            if args.debug: print(f'  Will generate model for failed execution')
+            execs_to_model.append((idx, ex))
         else:
             stuck.append((opcode, idx, ex))
 
-    end = timer()
+    if len(execs_to_model) > 2 and args.solver_parallel:
+        with Pool(processes=args.solver_parallel_cores) as pool:
+            if args.debug: print(f'Spawning {len(execs_to_model)} parallel assertion solvers on {args.solver_parallel_cores} cores')
+            models = [m for m in pool.starmap(gen_model_from_sexpr, [(args, idx, ex.solver.sexpr()) for idx, ex in execs_to_model])]
 
-    passed = (normal > 0 and len(models) == 0 and len(stuck) == 0)
-    if passed:
-        passfail = color_good('[PASS]')
     else:
-        passfail = color_warn('[FAIL]')
+        models = [gen_model(args, idx, ex) for idx, ex in execs_to_model]
+
+    end = timer()
+
+    passed = (normal > 0 and any(m.model is not None for m in models) == 0 and len(stuck) == 0)
```

Comment:
there may exist `m` such that m.model is None but m.result is unknown.  in that case, i don't think we want to say it is passed.

btw, doesn't `any()` return boolean?  i mean, `False == 0` is true, but not easy to read for me.

Reply:
will rewrite, the expression is confusing

Reply:
should read better in 3fe2666